File: product-service/src/main/java/com/intexsoft/webshop/productservice/repository/VendorRepository.java
package com.intexsoft.webshop.productservice.repository;

import com.intexsoft.webshop.productservice.model.Vendor;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface VendorRepository extends JpaRepository<Vendor, Long> {
    Optional<Vendor> findByNameIgnoreCase(String name);
}

File: product-service-kt/src/main/kotlin/com/intexsoft/webshop/productservicekt/repository/VendorRepository.kt
package com.intexsoft.webshop.productservicekt.repository

import com.intexsoft.webshop.productservicekt.model.Vendor
import org.springframework.data.jpa.repository.JpaRepository

interface VendorRepository : JpaRepository<Vendor, Long> {
    fun findByNameIgnoreCase(name: String): Vendor?
}
